src/GafferScene/RenderController.cpp

```diff
@@ -344,7 +387,7 @@ class RenderController::SceneGraph
 			const bool parentTransformChanged = m_parent && ( m_parent->m_changedComponents & TransformComponent );
 			if( ( m_dirtyComponents & TransformComponent ) || parentTransformChanged )
 			{
-				if( updateTransform( controller->m_scene->transformPlug(), parentTransformChanged ) )
+				if( updateTransform( controller->m_scene->transformPlug(), parentTransformChanged, controller->m_motionBlurOptions ) )
```

Comment:
When I first saw this I wondered what the interaction between `m_transformSegments` and `controller->m_motionBlurOptions` was going to be, since we already used the latter in the computation of the former. Looks like we only use the `shutter` part of the options in `updateTransform()` though, so perhaps we should pass that alone?

Actually, since we already store `m_transformTimes`, can we just compute it earlier where we're currently updating `m_transformSegments`, and ditch `m_transformSegments` entirely? Seems like the dataflow is a bit more obvious that way potentially? Probably worth considering how this impacts the discussion around the interaction between hashes and samples as well.

Reply:
I've gone with your direction of ditching m_transformSegments in c060dee.  It has quite a lot of knock-on effects - most quite positive.

The one slightly weird part:  we now don't modify m_transformTimes in transformSamples - we instead just ignore the times when a single sample is returned ( ie if samples are all identical ).  This keeps m_transformTimes around so we can use it for future evaluations when things might not all be identical.  But not modifying m_transformTimes means we also need a pointer m_transformTimesOutput, which tracks whether we are using the times from m_parent.